equinox/nn/normalization.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Optional, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+_shape_t = Union[int, Tuple[int], List[int]]
+
+
+class LayerNorm(Module):
+    """Layer Normalization as described in https://arxiv.org/abs/1607.06450"""
+
+    normalized_shape: _shape_t = static_field()
+    eps: float = static_field()
+    elementwise_affine: bool = static_field()
+    weight: Array
+    bias: Array
+
+    def __init__(
+        self,
+        normalized_shape: _shape_t,
+        eps: float = 1e-5,
+        elementwise_affine: bool = True,
+        *,
+        key: "jax.random.PRNGKey"
+    ):
+        """**Arguments:**
+        - `normalized_shape`: Input shape.
+        - `eps`: Value added to denominator for numerical stability. Default: `1e-5`.
+        - `elementwise_affine`: Whether the module has learnable affine parameters. Default: `True`.
+        - `key`: Ignored; provided for compatibility with the rest of the Equinox API.
+            (Keyword only argument.)
+        """
+        self.normalized_shape = normalized_shape
+        self.eps = eps
+        self.elementwise_affine = elementwise_affine
+        self.weight = jnp.ones(self.normalized_shape)
+        self.bias = jnp.zeros(self.normalized_shape)
```

Comment:
Maybe it's best to assign `self.weight = None` (and `bias`) if `elementwise_affine` is False? Save some memory.